internal/pkg/v2/README.md

```diff
@@ -0,0 +1,275 @@
+
+#### Single Configuration-Driven Executable
+
+The design supports a single executable capable of supporting endpoints from one or more of today's existing services.  This could be hardcoded -- as it is in the current Fuji release.  It could also easily be configuration-driven -- a single executable could provide any or all existing functionality for all core, support, and system management services based on a configuration file.  This would simplify artifact creation and provide a more flexible deployment story.
+
+#### RequestEnvelope's Strategy Property
+
+The RequestEnvelope includes a strategy property; currently defined are "sync", "async-push", and "asynch-poll".  The strategy property is intended to provide a transport-agnostic way of specifying the desired processing strategy to be applied to a use-case request.
+
+The "sync" option is implemented by default.  It processes the use-case request synchronously and returns a result only when processing has been completed.
+
+The "asynch-push" option provides for asynchronous processing of a use-case request.  The caller supplies an address as part of the request.  The request is accepted for processing and an intermediate result is immediately returned.  When request processing has asynchronously completed, the result will be pushed to the address provided.  This option has not been implemented and requires further definition and refinement.
+
+The "async-poll" option provides for asynchronous processing of a use-case request.  The request is accepted for processing and a unique token is immediately returned.  When request processing has asynchronously completed, the result will be stored for some finite timeframe.  The caller would provide the unique token to query for the result through a yet to be defined endpoint/process.  This option has not been implemented and requires further definition and refinement.
+
+
+
+### Known Open Issues Still To Be Addressed
+
+1. An example of a service-specific endpoint needs to be implemented.  This will illustrate how controllers for HTTP GET and DELETE method-based endpoints will create use-case request DTOs from HTTP headers and/or URL content.  It will also illustrate the approach to persistence, domain models, and how persistence schema and DTO definition diverge.
+2. The APIv2 specifications are inconsistent.  Certain definitions related to the ping, metrics, configuration, and version endpoints are accurate.  The rest of the document is not.  I intend to update the documents to make them consistent after the service-specific endpoint example has been completed, reviewed, and merged.
+3. Creation of an ADR to document the design, architecture, and approach to the APIv2 implementation.  I intend to repurpose a good portion of this document to create the ADR after the service-specific endpoint example has been completed, reviewed, and merged.
+4. go-mod-core-contracts integration.
```

Comment:
Mike and were discussing that the DTOs belong in go-mod-core-contracts. Thoughts?

Reply:
Eventually, yes.  At least a copy of them as they constitute the contract.